scenarioo-client/app/step/step.controller.js

```diff
@@ -335,9 +367,126 @@ function StepController($scope, $routeParams, $location, StepResource, HostnameA
         }
 
         var selected = SelectedBranchAndBuildService.selected();
-        return HostnameAndPort.forLink() + 'rest/branch/' + selected.branch + '/build/' + selected.build + '/usecase/' + $scope.stepIdentifier.usecaseName + '/scenario/' + $scope.stepIdentifier.scenarioName + '/image/' + imageName;
+
+        $scope.screenShotUrl = HostnameAndPort.forLink() + 'rest/branch/' + selected.branch + '/build/' + selected.build + '/usecase/' + $scope.stepIdentifier.usecaseName + '/scenario/' + $scope.stepIdentifier.scenarioName + '/image/' + imageName;
+    }
+
+    // This URL is only used internally, not for sharing
+    function initScreenshotURLs(){
+        initComparisonScreenshotUrl ();
+        initDiffScreenShotUrl();
+    }
+
+    function initComparisonScreenshotUrl () {
+        $scope.comparisonScreenShotUrl = ScreenshotUrlService.getComparisonScreenShotUrl($scope.comparisonBranchName, $scope.comparisonBuildName, $scope.stepIdentifier.usecaseName, $scope.stepIdentifier.scenarioName, $scope.comparisonScreenshotName);
+        $scope.comparisonScreenShotDescription = $scope.branch;
+    }
+
+    function initBaseBuildName() {
+        BranchesAndBuildsService.getDisplayNameForBuildName(SelectedBranchAndBuildService.selected().branch, SelectedBranchAndBuildService.selected().build, false).then(function(result){
```

Comment:
Agreed. We need something like a global appcontext and some events, when something changes. We also need to make sure when to invalidate our app context (e.g. any configuration changes).